database/deployment.go

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/go-vela/types/library"
+	"github.com/go-vela/types/raw"
+	"github.com/lib/pq"
+)
+
+var (
+	// ErrEmptyDeploymentNumber defines the error type when a
+	// Deployment type has an empty Number field provided.
+	ErrEmptyDeploymentNumber = errors.New("empty deployment number provided")
+
+	// ErrEmptyDeploymentRepoID defines the error type when a
+	// Deployment type has an empty RepoID field provided.
+	ErrEmptyDeploymentRepoID = errors.New("empty deployment repo_id provided")
+)
+
+// Deployment is the database representation of a deployment for a repo.
+type Deployment struct {
+	ID          sql.NullInt64      `sql:"id"`
+	Number      sql.NullInt64      `sql:"number"`
+	RepoID      sql.NullInt64      `sql:"repo_id"`
+	URL         sql.NullString     `sql:"url"`
+	User        sql.NullString     `sql:"user"`
+	Commit      sql.NullString     `sql:"commit"`
+	Ref         sql.NullString     `sql:"ref"`
+	Task        sql.NullString     `sql:"task"`
+	Target      sql.NullString     `sql:"target"`
+	Description sql.NullString     `sql:"description"`
+	Payload     raw.StringSliceMap `sql:"payload"`
+	Builds      pq.StringArray     `sql:"builds" gorm:"type:varchar(50)"`
+}
```

Comment:
suggestion: we might want to add Created as we do in other tables so we can more easily purge these

Reply:
added!